pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AbstractAnyTypeDeclaration.java

```diff
@@ -26,16 +27,52 @@
     }
 
 
-    /**
-     * Returns true if this type declaration is nested inside an interface, class or annotation.
-     */
     @Override
     public final boolean isNested() {
         return jjtGetParent() instanceof ASTClassOrInterfaceBodyDeclaration
                 || jjtGetParent() instanceof ASTAnnotationTypeMemberDeclaration;
     }
 
 
+    /**
+     * Returns true if the enclosing type of this type declaration
+     * is any of the given kinds. If this declaration is a top-level
+     * declaration, returns false. This won't consider anonymous classes
+     * until #905 is tackled. TODO 7.0.0
+     *
+     * @param kinds Kinds to test
+     */
+    // TODO 7.0.0 move that up to ASTAnyTypeDeclaration
+    public final boolean enclosingTypeIsA(TypeKind... kinds) {
+
+        if (!isNested()) {
+            return false;
+        }
+
+        Node parent = getNthParent(3);
+
+        for (TypeKind k : kinds) {
+            if (((ASTAnyTypeDeclaration) parent).getTypeKind() == k) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+
+    /**
+     * Returns the enclosing type of this type, if it is nested.
+     * Otherwise returns null. This won't consider anonymous classes
+     * until #905 is tackled. TODO 7.0.0
+     */
+    public final ASTAnyTypeDeclaration getEnclosingTypeDeclaration() {
+        if (!isNested()) {
+            return null;
+        }
+        return (ASTAnyTypeDeclaration) getNthParent(3);
```

Comment:
I'm a little concerned about this unchecked cast...

Reply:
Oh yeah you're right, I though isNested protected it but it doesn't check high enough in the tree